tests/generic_config_updater/test_dhcp_relay.py

```diff
@@ -0,0 +1,236 @@
+            cmds.append('config vlan dhcp_relay add {} 192.0.{}.{}'.format(vlan['vlan_id'], vlan['vlan_id'], i))
+    
+    duthost.shell_cmds(cmds=cmds)
+    time.sleep(5)
+    pytest_assert(duthost.is_service_fully_started('dhcp_relay'), "dhcp_relay service is not running during setup")
+
+@pytest.fixture(scope="module")
+def setup_vlan(duthosts, rand_one_dut_hostname, vlan_intfs_list, rand_vlan_port):
+    duthost = duthosts[rand_one_dut_hostname]
+
+    # --------------------- Setup -----------------------
+    try:
+        create_test_vlans(duthost, vlan_intfs_list, rand_vlan_port)
+        create_test_dhcp_servers(duthost, vlan_intfs_list)
+        
+    # --------------------- Testing -----------------------
+        yield
+
+    # --------------------- Teardown -----------------------
+    finally:
+        tearDown(duthost, vlan_intfs_list, rand_vlan_port)
+
+def tearDown(duthost, vlan_intfs_list, rand_vlan_port):
+    '''Clean up VLAN CONFIG for this test
+    '''
+    logger.info("VLAN test ending ...")
+    logger.info("Delete VLAN intf")
+
+    cmds = []
+
+    for vlan in vlan_intfs_list:
+        cmds.append('config vlan member del {} {}'.format(vlan['vlan_id'], rand_vlan_port))
+        cmds.append("config interface ip remove Vlan{} {}".format(vlan['vlan_id'], vlan['ip'].upper()))
+        cmds.append('config vlan del {}'.format(vlan['vlan_id']))
+
+    logger.info("Commands: {}".format(cmds))
+    duthost.shell_cmds(cmds=cmds,  module_ignore_errors=True)
+
+    # Clean up files
+    duthost.shell('rm -rf {}'.format(DUT_EMPTY_JSON_FILE))
+    duthost.shell('rm -rf {}'.format(DUT_ADD_SERVERS_JSON_FILE))
+    duthost.shell('rm -rf {}'.format(DUT_DEL_SERVERS_JSON_FILE))
+
+DUT_EMPTY_JSON_FILE='/tmp/dhcp_empty.json'
+def test_dhcp_relay_tc1_empty(duthost, setup_vlan):
+    """Test Empty JSON file to see if apply-patch command work
+
+    This test generates no change for the whole vlan config
+
+    If apply-patch succeeds, output show shows below(The output may look different except the last line)
+    admin@sonic:~$ sudo config apply-patch /tmp/dhcpempty.json
+    Patch Applier: Sorting patch updates.
+    Patch Applier: The patch was sorted into 0 changes.
+    Patch Applier: Applying changes in order.
+    Patch Applier: Verifying patch updates are reflected on ConfigDB.
+    Patch Applier: Patch application completed.
+    Patch applied successfully.
+    """
+    empty_json = []
+    duthost.copy(content=json.dumps(empty_json, indent=4), dest=DUT_EMPTY_JSON_FILE)
+
+    cmds = 'config apply-patch {}'.format(DUT_EMPTY_JSON_FILE)
+
+    logger.info("Commands: {}".format(cmds))
+    output = duthost.shell(cmds)
+
+    pytest_assert("Patch applied successfully" in output['stdout'], "apply-patch is not working correctly")
+
+DUT_ADD_SERVERS_JSON_FILE='/tmp/dhcp_add_servers.json'
+DUT_DEL_SERVERS_JSON_FILE='/tmp/dhcp_del_servers.json'
+def test_dhcp_relay_tc2_servers(duthost, setup_vlan):
+    """Test add and del dhcp servers cases
```

Comment:
>add [](http://example.com/codeflow?start=12&length=3)

There is some corner cases for "add":
1. reach maximal servers
2. add an existing one

Reply:
Will ask see if there is the design

Reply:
Low priority test case.